codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/types/StructureGenerator.java

```diff
@@ -90,13 +96,26 @@ public void render() {
         writeRbs(writer -> {
             String shapeName = symbolProvider.toSymbol(shape).getName();
             writer
-                    .openBlock("class $L < ::Struct[untyped]", shapeName)
+                    .openBlock("class $L", shapeName)
                     .write("include $L", Hearth.STRUCTURE)
                     .call(() -> {
                         shape.members().forEach(memberShape -> {
-                            String name = symbolProvider.toMemberName(memberShape);
-                            Symbol target = symbolProvider.toSymbol(model.expectShape(memberShape.getTarget()));
-                            writer.write("attr_accessor $L (): $L", name, target.getProperty("rbsType").get());
+                            String memberName = symbolProvider.toMemberName(memberShape);
+                            Shape target = model.expectShape(memberShape.getTarget());
+                            Symbol symbol = symbolProvider.toSymbol(target);
+                            String rbsType;
+                            // TODO: not sure if any of this belongs here
+                            if (target.hasTrait(StreamingTrait.class)) {
+                                // TODO: determine if input or output and set accordingly
```

Comment:
This is tricky to do I think and a shape can in theory be used by both input and output, but I think one of the knowledge indexes might let us do it?